src/db.rs

```diff
@@ -131,6 +132,17 @@ impl DB {
                         }
                     };
 
+                    if merge {
+                        // If the path exists in the database, add the ranks and set the epoch to
+                        // the largest of the parsed epoch and the already present epoch.
+                        if let Some(dir) = self.dirs.iter_mut().find(|dir| dir.path == path_str) {
+                            dir.rank += rank;
+                            dir.last_accessed = i64::max(epoch, dir.last_accessed);
+
+                            continue;
```

Comment:
I feel it would improve code clarity if there was an `else` clause rather than a `continue`.

Reply:
The only reason I didn't do that was to reduce duplicate code -- otherwise there would be two instances of `self.dirs.push(Dir {`: one for when `merge` is active and the path wasn't already stored, and one for when `merge` is inactive.

Maybe I could add a comment above the `continue`, explaining why? Or, if you insist, I could add an `else` block.

Reply:
For reference, here's what would need to be done (unless I overlooked something, which is exceedingly likely):

<details><summary>diff</summary>

```diff
diff --git a/src/db.rs b/src/db.rs
index 7041b4f..c3e8c6f 100644
--- a/src/db.rs
+++ b/src/db.rs
@@ -135,21 +135,30 @@ impl DB {
                     if merge {
                         // If the path exists in the database, add the ranks and set the epoch to
                         // the largest of the parsed epoch and the already present epoch.
-                        if let Some(dir) = self.dirs.iter_mut().find(|dir| dir.path == path_str) {
-                            dir.rank += rank;
-                            dir.last_accessed = Epoch::max(epoch, dir.last_accessed);
-
-                            continue;
-                        };
+                        match self.dirs.iter_mut().find(|dir| dir.path == path_str) {
+                            Some(dir) => {
+                                dir.rank += rank;
+                                dir.last_accessed = Epoch::max(epoch, dir.last_accessed);
+                            }
+                            None => {
+                                // FIXME: When we switch to PathBuf for storing directories inside Dir, just
+                                // pass `PathBuf::from(path_str)`
+                                self.dirs.push(Dir {
+                                    path: path_str.to_string(),
+                                    rank,
+                                    last_accessed: epoch,
+                                });
+                            }
+                        }
+                    } else {
+                        // FIXME: When we switch to PathBuf for storing directories inside Dir, just
+                        // pass `PathBuf::from(path_str)`
+                        self.dirs.push(Dir {
+                            path: path_str.to_string(),
+                            rank,
+                            last_accessed: epoch,
+                        });
                     }
-
-                    // FIXME: When we switch to PathBuf for storing directories inside Dir, just
-                    // pass `PathBuf::from(path_str)`
-                    self.dirs.push(Dir {
-                        path: path_str.to_string(),
-                        rank,
-                        last_accessed: epoch,
-                    });
                 }
                 [] | [""] => {} // ignore blank lines
                 line => {
```
</details>

Note the two instances of `self.dirs.push(Dir {` now. We need to handle the case where the directory doesn't exist in the database, which was originally done by just letting the `if let Some(...` fall through.

Reply:
You're right, I missed that.